The build environment was not happy with this:
``` 
12:28:36 mkdir -p /sd/workspace/src/github.com/vespa-engine/vespa/build/go && GOTMPDIR=/sd/workspace/src/github.com/vespa-engine/vespa/build/go go test ./...
12:28:36 go: go.mod requires go >= 1.22 (running go 1.21.9; GOTOOLCHAIN=local)
12:28:36 make: *** [Makefile:140: test] Error 1
```

--- a/client/go/Makefile
+++ b/client/go/Makefile
@@ -113,8 +113,10 @@ install-all: all manpages
 
 ci:
 ifdef CI
-# Ensure that CI systems use a proxy for downloading dependencies
+# Ensure that CI systems use a proxy for downloading dependencies, and
+# automatically downloads the necessary toolchain
 	go env -w GOPROXY="https://proxy.golang.org,direct"
+	go env -w GOTOOLCHAIN="auto"
 	go env
 endif
 

--- a/client/go/go.mod
+++ b/client/go/go.mod
@@ -1,6 +1,6 @@
 module github.com/vespa-engine/vespa/client/go
 
-go 1.22
+go 1.22.0
 
 require (
 	github.com/alessio/shellescape v1.4.2

--- a/client/go/go.mod
+++ b/client/go/go.mod
@@ -1,6 +1,6 @@
 module github.com/vespa-engine/vespa/client/go
 
-go 1.22.0
+go 1.22.4
 
 require (
 	github.com/alessio/shellescape v1.4.2

